plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -359,6 +387,164 @@ class CodeTransformChatController(
         }
     }
 
+    private suspend fun hilTryResumeAfterError(errorMessage: String) {
+        codeTransformChatHelper.addNewMessage(buildHilErrorContent(errorMessage))
+        codeTransformChatHelper.addNewMessage(buildHilResumeWithErrorContent())
+
+        try {
+            codeModernizerManager.rejectHil()
+            runInEdt {
+                codeModernizerManager.getBottomToolWindow().show()
+            }
+
+            codeTransformChatHelper.chatDelayLong()
+
+            codeModernizerManager.resumePollingFromHil()
+        } catch (e: Exception) {
+            codeTransformChatHelper.updateLastPendingMessage(buildHilCannotResumeContent())
```

Comment:
BLocker: We should we have error telemetry reporting here. Or somewhere in a `catch` statement to capture a runtime error state for HIL.

```
            telemetry.logHil(
                CodeModernizerSessionState.getInstance(context.project).currentJobId?.id as String,
                HilTelemetryMetaData(
                    cancelledFromChat = false,
                    reason = "Runtime Error",
                    result = "FAILURE",
                )
            )
            ```

Reply:
We can add generic log details to idea logs in the same try/catch with more specific info with `logger.error (e.message)`